src/openrct2/peep/peep.c

```diff
@@ -10419,10 +10420,10 @@ static sint32 sub_693C9E(rct_peep *peep)
 			continue;
 		if (top_z < mapElement->base_height)
 			continue;
+		if ((mapElement->flags & MAP_ELEMENT_FLAG_GHOST))
```

Comment:
@IntelOrca isn't this a bit buggy here?

If the ghost is also the last element, it will `continue` here and break out of `do { … } while (!map_element_is_last_for_tile(mapElement++));` loop and continue pathfinding as if it was a valid element.

I think there should be a check prior to this one for `ghost` **and** `last` flags being set and `return 1` in such case.